---
title: "Phage microbiome analysis"
author: "Kristi Gdanetz MacCready"
date: "2/23/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE)
knitr::opts_knit$set(root.dir = "~/Documents/Flower_Microbiome/")
```

# Load libraries 
```{r, warning=FALSE, message=FALSE}
library(vegan)
library(tidyverse)
library(readxl)
library(phyloseq)
library(cowplot)
theme_set(theme_cowplot())
set.seed(4)
source("~/Documents/GitHub/DenefLab-MicrobeMiseq/R/miseqR.R")
```

### Colors
* 3 phage mix = ED1E24 (red)
* Cot4 = 70C6A5 (blue/green)
* Shl2 = #8750A0 (mauve)
* Rec1 =  9ACA3C (yellow green)
* Water = 426FB6 (dk blue)

* Agriphage = FAA51A (orange)
* Water = 426FB6 (dk blue) 

* Cherry and 100% apple
** Before = #ED187A (pink)
** 16h = #71C054 (green)
** 40h = #5B51A3 (violet)

* 25% apple 
** Before = #F7AEC0 (lt pink)
** 16h = #C5E2B5 (ltgreen)
** 40h =  #afa9d3 (lt violet)

# Import data

### Taxonomy table
RDP taxonomy
```{r}
rdp <- read_delim(file = "raw_data/taxonomy_phage_16s_otus_RDP_edited.txt", delim = "\t", 
                     col_names = TRUE, na = c("NA", " ", ""), show_col_types = FALSE) %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus) %>%
  column_to_rownames(var = "OTU")
```

SINTAX taxonomy
```{r}
sintax <- read_delim(file = "raw_data/taxonomy_phage_16s_otus_edited.sintax", delim = "\t", 
                     col_names = TRUE, na = c("NA", " ", ""), show_col_types = FALSE) %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus)
```

Add OTUs sums to taxonomy table
```{r}
rdp2 <- shared_sum %>% 
  # format data frame for calculations 
  column_to_rownames(var = "Sample_ID") %>%
  select(-RowSum) %>% 
  t() %>% as.data.frame() %>%
  mutate(OTUSum = rowSums(.,)) %>%
  rownames_to_column(var = "OTU_ID") %>%
  # join with taxonomy information 
  select(OTU_ID, OTUSum) %>%
  as.data.frame() %>% arrange(desc(OTUSum)) %>%
  left_join(rownames_to_column(rdp, var = "OTU_ID"), by = "OTU_ID")
```

Bad OTUS - plants and chloroplasts 
```{r}
chloros <- filter(rdp2, Phylum == "Plantae" | Family == "Chloroplast")
```

### Shared table
```{r}
shared <- read_delim(file = "raw_data/phage_16s_OTU_table.txt", delim = "\t", 
                     col_names = TRUE, show_col_types = FALSE) %>%
  column_to_rownames(var = "#OTU ID") 
```

### Metadata
```{r}
meta <- read_excel(path = "raw_data/phage_metadata.xlsx", 
                   col_names = TRUE, na = c("NA", " ", ""), 
                   col_types = c("numeric", "text", "text", "text", 
                                 "text", "text", "text", "date",
                                 "text", "text", "numeric", "text",
                                 "text", "text", "text", "text", 
                                 "text", "text", "text", "text") ) %>%
  select(-N, -Plate, -Well, -Sample_ID, -read1, -read2) %>%
  mutate(Growth_stage = factor(Growth_stage, ordered = TRUE,
                               levels = c("25% Bloom", "100% Bloom", "Bloom") ),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("before", "0h", "6h", "16h", "24h", "40h", "48h")),
         Stage2 = paste(Growth_stage, Timepoint, sep = " "),
         Stage2 = factor(Stage2, ordered = TRUE,
                         levels = c("25% Bloom 0h", "25% Bloom 16h","25% Bloom 40h",
                                    "100% Bloom 16h", "100% Bloom 40h",
                                    "Bloom 16h", "Bloom 24h", "Bloom 40h", "Bloom 48h")) )
```

### Group replicate samples
Join meta and shared, sum replicates, make new sample labels 
```{r}
meta_share <- shared %>%
  # join shared and metadata tables
  t() %>% as.data.frame() %>%
  rownames_to_column(var = "Seq_ID") %>%
  left_join(meta, by = "Seq_ID") %>%
  # sum across replicate samples
  select(Seq_ID, Submission_ID, Year, Host, Location, Treatment, 
         Growth_stage, Stage2, Hours_post_spray, Timepoint, 
         Tree, Tissue, Replicate, Notes, starts_with("OTU")) %>%
  filter(Notes != "TOSS") %>% 
  group_by(Year, Host, Location, Treatment, Growth_stage, Stage2, Hours_post_spray, Timepoint, Tree, Tissue) %>%
  summarise( across( c(starts_with("OTU")), ~ sum(.x, na.rm = TRUE) ) )  %>%
  ungroup() %>%
  # make new sample IDs
  mutate(tmp1 = recode(Year, "2020" = "20", "2021" = "21"),
         tmp2 = recode(Host, "Apple" = "A", "Cherry" = "C"),
         tmp5 = paste(tmp1, tmp2, sep= ""),
         tmp3 = recode(Growth_stage,
                       "25% Bloom" = "25%", "100% Bloom" = "100%",
                       "Bloom" = "100%"),
         tmp4 = recode(Tissue, 
                       ANTH = "A",  Flower = "F",
                       RECEP = "R",  Stigma = "S"), 
         tmp6 = paste(tmp4, Tree, sep = ""),
         Sample_ID = paste(tmp5, Location, Treatment, 
                           tmp3, Timepoint, tmp6, sep = "_")) %>%
  select(Sample_ID, everything(), starts_with("OTU"), -tmp1, -tmp2, -tmp3, -tmp4, -tmp5, -tmp6)

# new metadata data frame
meta2 <- select(meta_share, everything(), -starts_with("OTU")) 
# new shared data frame 
shared2 <- select(meta_share, Sample_ID, starts_with("OTU")) 
```

```{r}
# get sample sums for apple 2021
apple2021 <- filter(meta2, Host == "Apple" & Year == "2021" & Location == "NWHRC") #45

ap21_sum_meta <- apple2021 %>%
  left_join(shared_sum, by = "Sample_ID") %>%
  select(Sample_ID, starts_with("OTU")) %>%
  pivot_longer(names_to = "OTUs", values_to = "Abundance", cols = c(starts_with("OTU")) ) %>% #36,000
  filter(!OTUs %in% chloros$OTU_ID) %>% #31,680 
  pivot_wider(names_from = "OTUs", values_from = "Abundance") %>% #45
  column_to_rownames(var = "Sample_ID") %>% 
  mutate(RowSum = rowSums(.,)) %>%
  rownames_to_column(var = "Sample_ID") %>% 
  right_join(apple2021, by = "Sample_ID") %>%
  dplyr::select(Sample_ID, Year, Host, Location, Treatment, 
                Growth_stage, Hours_post_spray, Timepoint, Tree, Tissue,  
                RowSum, starts_with("OTU"))

# make bar plot per sample
test_plot <- ap21_sum_meta %>%
   pivot_longer(names_to = "OTU_ID", values_to = "Abundance", cols = c(starts_with("OTU")) ) %>%
   left_join(rdp2, by = "OTU_ID") %>%
   ggplot(aes(x = Sample_ID, y = Abundance, fill = Genus)) +
   geom_bar(stat = "identity") + theme(legend.position = "none", axis.text.x = element_text(angle = -90))
test_plot
```

### Make phlyoseq object
```{r}
psq <- phyloseq(otu_table(column_to_rownames(shared2, var = "Sample_ID"), taxa_are_rows = FALSE),
                tax_table(as.matrix(rdp)),
                sample_data(column_to_rownames(meta2, var = "Sample_ID")))
psq
```

# Sample statistics
### Reads per OTU
```{r}
plot0 <- rdp2 %>%
  filter(OTUSum >= 100) %>%
  ggplot(data=, aes(x = reorder(OTU_ID, OTUSum) , y = OTUSum)) + 
  geom_col() + scale_y_continuous(limits = c(0, 100000)) +
  theme(axis.text.x = element_text(angle=-90))
plot0
```

### Reads per sample
Calculate reads per sample and plot
```{r}
# Summary statistics on read counts 
min(sample_sums(psq)) #3
mean(sample_sums(psq)) #9,861
max(sample_sums(psq))  #36,474

# Get sum of reads per sample
shared_sum <- column_to_rownames(shared2, var = "Sample_ID") %>% 
  mutate(RowSum = rowSums(.,)) %>%
  select(RowSum, starts_with("OTU")) %>%
  rownames_to_column(var = "Sample_ID") %>%
  as.data.frame() %>% arrange(desc(RowSum))
# Plot reads per sample
plot1 <- ggplot(data=shared_sum, aes(x = reorder(Sample_ID, RowSum) , y = RowSum)) + geom_col()
plot1
```

Abundance/composition of mock and negative control samples
```{r}
# list of control samples
L1 <- meta2 %>% filter(Treatment == "Neg" | Treatment == "MOCK" | Treatment == "resub") 
# plot control samples 
plot2 <- shared_sum %>%
  filter(Sample_ID %in% L1$Sample_ID) %>%
  ggplot(aes(x = reorder(Sample_ID, RowSum) , y = RowSum)) + 
  geom_col() #+ theme(axis.text.x = element_text(angle=-90))
plot2
```

Read abundance in cherry samples
```{r}
# list of control samples
L2 <- filter(meta2, Host == "Cherry") 
# plot control samples 
plot3 <- shared_sum %>%
  inner_join(L2, by = "Sample_ID") %>%
  ggplot(aes(x = reorder(Sample_ID, RowSum), y = RowSum, fill = Timepoint)) + 
  geom_col() + theme(axis.text.x = element_text(angle=-90))
plot3
```

Read abundance in apple samples
```{r}
# list of control samples
L3 <- filter(meta2, Host == "Apple") 
# plot control samples 
plot4 <- shared_sum %>%
  inner_join(L3, by = "Sample_ID") %>%
  ggplot(aes(x = reorder(Sample_ID, RowSum), y = RowSum, fill = Timepoint)) + 
  geom_col() + theme(axis.text.x = element_blank())
plot4
```

Read abundance in phage treated samples
```{r}
# list of control samples
L4 <- filter(meta2, Treatment != "Water") 
# plot control samples 
plot5 <- shared_sum %>%
  inner_join(L4, by = "Sample_ID") %>%
  ggplot(aes(x = reorder(Sample_ID, RowSum), y = RowSum, fill = Timepoint)) + 
  geom_col() + theme(axis.text.x = element_blank())
plot5
```

Read abundance in water treated samples
```{r}
# list of control samples
L5 <- filter(meta2, Treatment == "Water") 
# plot control samples 
plot6 <- shared_sum %>%
  inner_join(L5, by = "Sample_ID") %>%
  ggplot(aes(x = reorder(Sample_ID, RowSum), y = RowSum, fill = Timepoint)) + 
  geom_col() + theme(axis.text.x = element_blank())
plot6
```

# Bar & Area plots
### phyloseq object
Make new phyloseq object just for bar plots. Remove chloroplasts, minimum read counts, minimum sample occurrence. Group replicate trees from each tissue+treatment+timepoint combination (replicate samples already grouped). 
```{r}
# extract sample names for shared table, sum OTU counts
shared3 <- shared2 %>%
  separate(Sample_ID, into = c("sample_name", "tis+rep"), sep="_(?=[^_]+$)") %>% 
  #do I need to account for different tissues here?
  select(-"tis+rep") %>% 
  group_by(sample_name) %>%
  summarise( across( c(starts_with("OTU")), ~ sum(.x, na.rm = TRUE) ) )  %>%
  ungroup()
# extract sample names for meta data
meta3 <- meta2 %>% 
  separate(Sample_ID, into = c("sample_name", "tis+rep"), sep="_(?=[^_]+$)") %>%
  #do I need to account for different tissues here?
  select(-"tis+rep", -Tree) %>% 
  distinct(sample_name, .keep_all = TRUE)
```

```{r}
psq_bar <- phyloseq(otu_table(column_to_rownames(shared3, var = "sample_name"), taxa_are_rows = FALSE),
                    tax_table(as.matrix(rdp)),
                    sample_data(column_to_rownames(meta3, var = "sample_name")))
psq_bar
```

### Genera abundance table
```{r}
gen_tab <- psq_bar %>%
  # filter for apple samples for manuscript
  subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>%
  subset_samples(Host == "Apple") %>%
  subset_samples(Location == "NWHRC" | Location == "CT") %>%
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>%
  tax_glom("Genus") %>% 
  # calculate relative abundance
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%
  # average relative abundance across replicates
  group_by(Genus, Location, Treatment, Growth_stage, Timepoint) %>%
  summarise(Avg_abund = round(mean(Abundance, na.rm = TRUE), digits = 4) ) %>%
  # format table for genera and sort samples
  filter(Genus == "Erwinia" | Genus == "Pseudomonas" | Genus == "Pantoea") %>%
  arrange(Location, Treatment, Growth_stage, Timepoint)
```

Abundances at 0h timepoint
```{r}
# table with repliciate samples kept separate
gen_rep_tab <- psq %>%
  # filter for apple samples for manuscript
  subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>%
  subset_samples(Host == "Apple") %>%
  subset_samples(Location == "NWHRC" | Location == "CT") %>%
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>%
  tax_glom("Genus") %>% 
  # calculate relative abundance
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%
  # format table for genera and sort samples
  filter(Genus == "Erwinia" | Genus == "Pseudomonas" | Genus == "Pantoea",
         Timepoint == "0h") %>%
  select(Sample, Abundance, Treatment, Stage2, Tree, Genus) %>% 
  arrange(Treatment, Stage2, Genus, Tree) 

### do statistics on abundance of individual genera between Treatments
# Erwinia
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Erwinia")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Erwinia")$Abundance,
       alternative = "two.sided") #alternative = "less" #p-value = 0.1131
# Pantoea
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Pantoea")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Pantoea")$Abundance,
       alternative = "two.sided") #p-value = 0.3618
# Pseudomonas
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Pseudomonas")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Pseudomonas")$Abundance,
       alternative = "two.sided") #p-value = 0.4383
```

Abundances at end of experiment
```{r}
# phyloseq object with replicate samples kept separate
gen_rep_tab <- psq %>%
  # filter for apple samples for manuscript
  subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>%
  subset_samples(Host == "Apple") %>%
  subset_samples(Location == "NWHRC") %>%
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>%
  tax_glom("Genus") %>% 
  # calculate relative abundance
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%
  # format table for genera and sort samples
  filter(Genus == "Erwinia" | Genus == "Pseudomonas" | Genus == "Pantoea",
         Stage2 == "100% Bloom 40h") %>%
  select(Sample, Abundance, Treatment, Stage2, Tree, Genus) %>% 
  arrange(Treatment, Stage2, Genus, Tree) 

### do statistics on abundance of individual genera between Treatments
# Erwinia
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Erwinia")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Erwinia")$Abundance,
       alternative = "two.sided") #p-value = 0.9493
# Pantoea
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Pantoea")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Pantoea")$Abundance,
       alternative = "two.sided") #p-value = 0.9117
# Pseudomonas
t.test(x = filter(gen_rep_tab, Treatment == "Agriphage", Genus == "Pseudomonas")$Abundance,
       y = filter(gen_rep_tab, Treatment == "Water", Genus == "Pseudomonas")$Abundance,
       alternative = "two.sided") #p-value = 0.8632
```

### Genus level
get list of low abundant genera to group into "Other" 
```{r eval=FALSE, include=FALSE}
# summerise phyloseq object by genus names 
pq_gen_l <- group_by(psq_bar_g, Genus) %>% summarise(Total = sum(Abundance), 
                                                 N= n())
# Less than 0.1 % 
L2 <- filter(pq_gen_l, Total < 0.001) #252 genera
# Less than 1 %
L2 <- filter(pq_gen_l, Total < 0.01) #361 genera, will use this one
v1 <- as.vector(L2$Genus)
# greater than 0.1 %
L3 <- filter(pq_gen_l, Total >= 0.001) #143 genera

#make a vector for Other
v2 <- rep("Other", times = 361)
```

Make phloseq object into long format data frame for ggplot 
```{r}
psq_bar_g <- psq_bar %>%
  subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>%
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>%
  tax_glom(taxrank = "Genus") %>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%
  # replace names in v1 with values in v2 (Other)
  mutate(Genus = stringr::str_replace_all(Genus, setNames(v2, v1)))
  # remove low-abundance taxa = <5 reads per sample
  #physq.bac = prune_taxa(taxa_sums(physq.bac) > 5, physq.bac)
psq_bar_g
```

### Apple 2020
```{r}
area1 <- psq_bar_g %>%
  filter(Host == "Apple", Year == "2020", Location == "CT") %>% 
  ggplot(aes(x = Hours_post_spray, y =  Abundance, fill = Genus, Treatment)) +
  geom_area() +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_vline(xintercept = 48, linetype = "dashed") +
  facet_grid(Treatment~.) +
  scale_y_continuous(name = "Relative Abundance (%)", expand=c(0,0),
                     breaks = c(0, 0.25, 0.5, 0.75, 1),
                     labels = c(0, 25, 50, 75, 100)) +
  scale_x_continuous(name = "Time (hrs)", breaks = c(0, 16, 40, 48, 64, 88) ) +
                     #breaks = c(16, 40, 48, 64, 88)) +
  scale_fill_manual(name = "Genus",
                     values = c("#ED1E24", "#F58565", #red
                                "#40B9EB", "#B8DDF5", #lt blue
                                "#FAA51A", "#FEC679", #orange
                                "#426FB6", "#AAB7DD", #dark blue
                                "#F4EB21", "#F8F18C", #yellow
                                "#5B51A3", "#afa9d3", #violet
                                "#9ACA3C", "#C2DD89", #yellow-green
                                "#8750A0", "#C4ACD3", #mauve
                                "#71C054", "#A9D595", #green
                                "#D0499A", "#DF92BF", #mauve-pink
                                "#70C6A5", "#AADAC6", #blue-green
                                "#ED187A", "#F7AEC0", #pink
                                "#ED1E24", "#F58565", #red
                                "#40B9EB", "#B8DDF5", #lt blue
                                "#FAA51A", "#FEC679", #orange
                                "#426FB6", "#AAB7DD", #dark blue
                                "#F4EB21", "#F8F18C", #yellow
                                "#5B51A3", "#afa9d3", #violet
                                "#9ACA3C", "#C2DD89", #yellow-green
                                "#8750A0", "#C4ACD3", #mauve
                                "#71C054", "#A9D595", #green
                                "#D0499A", "#DF92BF", #mauve-pink
                                "#70C6A5", "#AADAC6", #blue-green
                                "#ED187A", "#F7AEC0" ),  #pink
                    guide = guide_legend(ncol=1)) +
  theme(axis.text = element_text(size = 8), #reduce text size
        strip.text = element_text(size = 8), axis.title = element_text(size = 10),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(2.0, "mm"), #reduce legend size
        legend.justification =  "top",
        plot.title = element_text(size=12)) #+ggtitle("Apple 2020 CT")
area1
```

### Apple 2021
Plot relative abundance across timepoints 
```{r}
area2 <- psq_bar_g %>%
  filter(Host == "Apple", Year == "2021", Location == "NWHRC") %>% #Hours_post_spray >= 16
  ggplot(aes(x = Hours_post_spray, y =  Abundance, fill = Genus, Treatment)) +
  geom_area() + 
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_vline(xintercept = 88, linetype = "dashed") +
  facet_grid(Treatment~.) +
  scale_y_continuous(name = "Relative Abundance (%)",
                     breaks = c(0, 0.25, 0.5, 0.75, 1), expand=c(0,0),
                     labels = c(0, 25, 50, 75, 100)) +
  scale_x_continuous(name = "Time (hrs)",
                     breaks = c(-8.5, 0, 16, 40, 88, 112, 136),labels = c(-9, 0, 16, 40, 88, 112, 136) ) +
                     #breaks = c(16, 40, 88, 112, 136),labels = c(16, 40, 88, 112, 136)) +
  scale_fill_manual(name = "Genus",
                     values = c("#ED1E24", "#F58565", #red
                                "#40B9EB", "#B8DDF5", #lt blue
                                "#FAA51A", "#FEC679", #orange
                                "#426FB6", "#AAB7DD", #dark blue
                                "#F4EB21", "#F8F18C", #yellow
                                "#5B51A3", "#afa9d3", #violet
                                "#9ACA3C", "#C2DD89", #yellow-green
                                "#8750A0", "#C4ACD3", #mauve
                                "#71C054", "#A9D595", #green
                                "#D0499A", "#DF92BF", #mauve-pink
                                "#70C6A5", "#AADAC6", #blue-green
                                "#ED187A", "#F7AEC0", #pink
                                "#ED1E24", "#F58565", #red
                                "#40B9EB", "#B8DDF5", #lt blue
                                "#FAA51A", "#FEC679", #orange
                                "#426FB6", "#AAB7DD", #dark blue
                                "#F4EB21", "#F8F18C", #yellow
                                "#5B51A3", "#afa9d3", #violet
                                "#9ACA3C", "#C2DD89", #yellow-green
                                "#8750A0", "#C4ACD3", #mauve
                                "#71C054", "#A9D595", #green
                                "#D0499A", "#DF92BF", #mauve-pink
                                "#70C6A5", "#AADAC6", #blue-green
                                "#ED187A", "#F7AEC0" ),  #pink
                    guide = guide_legend(ncol=1)) +
  theme(axis.text = element_text(size = 8), #reduce text size
        strip.text = element_text(size = 8),axis.title = element_text(size = 10),
        legend.text = element_text(size = 8),  legend.title = element_text(size=10),
        legend.key.size = unit(2.5, "mm"), #reduce legend size
        legend.justification =  "top", plot.title = element_text(size=12)) #+ggtitle("Apple 2021 NWHRC")
area2 
```

### Class level
```{r}
psq_c <- psq %>%
  subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>%
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>%
  tax_glom(taxrank = "Class") %>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%
  group_by(Sample, Year, Host, Location, Treatment, 
           Growth_stage, Hours_post_spray, Phylum, Class) %>%
  summarise(Mean_rel_abund = round(mean(Abundance, na.rm = TRUE), digits = 5), 
            N = n())
psq_c
```


# Richness
### Format data 
Calculate values and make data frame
```{r}
# not normalized, remove empty taxa, remove host sequences, remove control samples 
psq_0 <- prune_taxa(taxa_sums(psq) > 0, psq) %>% #800 taxa 
  #subset_samples(Treatment != "MOCK" & Treatment != "Neg" & Treatment != "resub") %>% #170 samples
  subset_samples(Timepoint != "MOCK" & Timepoint != "Neg" & Timepoint != "resub" & Timepoint != "TOSS") %>% 
  #subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>% #649 taxa
  prune_taxa(taxa_sums(.) > 0, .) #796 taxa 

# format data frame with diversity indices for plotting
alpha1 <- estimate_richness(psq_0, split = TRUE,  # Calculate diversity indices
                               measures = c("Observed", "Shannon", "Chao1", "InvSimpson")) %>%
  # get sample IDs, remove X, replace period with percent 
  rownames_to_column(var = "tmp1") %>%
  separate(col = "tmp1", into = c("tmp2", "tmp3"), sep = "X", remove = FALSE) %>%
  mutate(tmp3 = str_replace_all(tmp3, pattern = "100.", replacement="100%"),
         tmp3 = str_replace_all(tmp3, pattern = "25.", replacement="25%"),
         tmp3 = str_replace_all(tmp3, pattern = "SHL2.T1", replacement="SHL2-T1"),
         Sample_ID = str_replace_all(tmp3, pattern = "3.Phage.mix", replacement="3 Phage mix")) %>%
  select(-starts_with("tmp"))

# add metadata to alpha diversity metrics
alpha2 <- left_join(meta2, alpha1, by = "Sample_ID") 

# Make long format for plotting
alpha3 <- pivot_longer(data = alpha2, 
                       cols = c("Observed", "Shannon", "Chao1", "InvSimpson"),
                       names_to = "Index", values_to = "Alpha", 
                       values_drop_na = FALSE) %>%
  mutate(Index = factor(Index, levels = c("Observed", "Shannon", "Chao1", "InvSimpson"), 
                        ordered = TRUE ))
```

### Statistical tests
##### Apple 2020
```{r}
# format data frame
testa20 <- filter(alpha3, Host == "Apple", Year == "2020", Location == "CT") %>%
  dplyr::select(Treatment, Growth_stage, Timepoint, Tree, Index, Alpha) %>%
  pivot_wider(names_from = c("Treatment", "Index"), values_from = "Alpha")

# anova of all timepoints - single treatment 
ANOVA_ap20 <- alpha3 %>%
  filter(Host == "Apple", Year == "2020", Location == "CT", Treatment == "Agriphage",
         #Index == "Shannon" #p= 0.938
         #Index == "Observed" #p= 0.551
         Index == "Chao1" #p= 0.404
         ) %>% 
  aov(Alpha~Hours_post_spray, data = .) 
summary(ANOVA_ap20) 

ANOVA_ap20W <- alpha3 %>%
  filter(Host == "Apple", Year == "2020", Location == "CT", Treatment == "Water",
         Index == "Shannon" #p= 0.219
         #Index == "Observed" #p= 0.447
         #Index == "Chao1" #p= 0.643
         ) %>% 
  aov(Alpha~Hours_post_spray, data = .) 
summary(ANOVA_ap20W) 
TukeyHSD(ANOVA_ap20W)

# 16h 25% bloom
var.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.9622
t.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.5379
var.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.3018
t.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.2317

# 40h 25% bloom
var.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.9406
t.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.0374
var.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.1897
t.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.001738

# 16h 100% bloom
var.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.7964
t.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.4594
var.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.8483
t.test(x = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.492

# 40h 100% bloom
var.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.6509
t.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.9963
var.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.1397
t.test(x = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa20, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.9596
```

##### Apple 2021
```{r}
# format data frame
testa21 <- filter(alpha3, Host == "Apple", Year == "2021", Location == "NWHRC") %>%
  dplyr::select(Treatment, Growth_stage, Timepoint, Tree, Index, Alpha) %>%
  pivot_wider(names_from = c("Treatment", "Index"), values_from = "Alpha")

# anova of all timepoints - single treatment 
ANOVA_ap21 <- alpha3 %>%
  filter(Host == "Apple", Year == "2021", Location == "NWHRC", Treatment == "Agriphage",
         Index == "Shannon" #p= 0.34
         #Index == "Observed" #p= 0.249
         #Index == "Chao1" #p= 0.29
         ) %>% 
  aov(Alpha~Hours_post_spray, data = .) 
summary(ANOVA_ap21) 

ANOVA_ap21W <- alpha3 %>%
  filter(Host == "Apple", Year == "2021", Location == "NWHRC", Treatment == "Water",
         Index == "Shannon" #p= 0.0532; 100% 40h-25% 40h
         #Index == "Observed" #p= 0.351
         #Index == "Chao1" #p= 0.611
         ) %>% 
  mutate(Hours_post_spray = factor(Hours_post_spray, ordered = TRUE,
                                   levels=c( -8.5, 16, 40, 112, 136),
                                   labels = c( "before", "25% 16h", "25% 40h", "100% 16h", "100% 40h") )) %>%
  aov(Alpha~Hours_post_spray, data = .) 
summary(ANOVA_ap21W) 
TukeyHSD(ANOVA_ap21W)

# before sample
var.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.7642
t.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.2221
var.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.1131
t.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.06288
var.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Chao1`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Chao1`,
            alternative = "two.sided") #p=0.7657
t.test(x = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Agriphage_Chao1`, 
            y = filter(testa21, Timepoint == "0h", Growth_stage == "25% Bloom")$`Water_Chao1`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.2479

# 16h 25% bloom
var.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.5472
t.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.6909
var.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.9151
t.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.4271

# 40h 25% bloom
var.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.08445
t.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.4464
var.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.0358
t.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "25% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = FALSE) #p=0.3294

# 16h 100% bloom
var.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=0.3257
t.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.2579
var.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=0.6725
t.test(x = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "16h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.9365

# 40h 100% bloom
var.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided") #p=
t.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Observed`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Observed`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.292
var.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided") #p=
t.test(x = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Agriphage_Shannon`, 
            y = filter(testa21, Timepoint == "40h", Growth_stage == "100% Bloom")$`Water_Shannon`,
            alternative = "two.sided", paired = FALSE, var.equal = TRUE) #p=0.28
```

### Plots 
##### Apple 2020
Plots for apple; time on x-axis
```{r}
# get values for error bars
line_ap20 <- alpha3 %>% 
  filter(Host == "Apple", Year == "2020", Location == "CT" ) %>% 
  mutate(Hours_post_spray = factor(Hours_post_spray, levels=c(16, 40, 64, 88),
                                    labels = c("16", "40", "64", "88"), ordered = TRUE)) %>%
  group_by(Year, Host, Treatment, Timepoint, Hours_post_spray, Index) %>%
  summarise(group_mean = mean(Alpha, na.rm = TRUE), 
            group_sd = sd(Alpha, na.rm = TRUE)) %>% ungroup()

# make the plot 
alpha_app20 <- alpha3 %>%
  filter(Host == "Apple", Year == "2020", Location == "CT", 
         Timepoint != "MOCK", Timepoint != "Neg", Timepoint != "resub", Timepoint != "TOSS") %>%
  mutate(Hours_post_spray = factor(Hours_post_spray, levels=c(16, 40, 64, 88),
                                   labels = c("16", "40", "64", "88"), ordered = TRUE)) %>%
  ggplot(aes(x = Hours_post_spray, y = Alpha, color = Treatment)) +
  facet_grid(Index~., scales = "free") + 
  geom_errorbar(data=line_ap20, aes(x=Hours_post_spray, y=group_mean, 
                ymin=group_mean-group_sd, ymax=group_mean+group_sd), 
                position = "dodge") +
  geom_crossbar(data=line_ap20, aes(x=Hours_post_spray, y=group_mean,  
                ymin=group_mean, ymax=group_mean),
                position = "dodge", fatten = 2) + 
  geom_jitter(position = position_jitterdodge(jitter.width=0.1), size = 2) +
  scale_color_manual(name = "Treatment",
                     values = c("#FAA51A", "#426FB6")) +
  scale_x_discrete(name = "Timepoint",
                   breaks = c("16", "40", "64", "88"),
                   labels = c("25% 16h","25% 40h", "100% 16h", "100% 16h")) +
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 10),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(3, "mm"), legend.position = "bottom") +
  ylab("Alpha Diversity Metric") + ggtitle("Apple 2020")
alpha_app20

# line plot
line_app20 <- line_ap20 %>%
  ggplot(aes(x = Hours_post_spray, y = group_mean, 
             color = Treatment, group = Treatment)) +
  facet_grid(Index~., scales = "free") + 
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_errorbar(aes(x=Hours_post_spray, y=group_mean, 
                    ymin=group_mean-group_sd, ymax=group_mean+group_sd),
                width = 0.5) +
  geom_line(size=1.5) + geom_point(size=2) +
  scale_color_manual(name = "Treatment",
                     values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 10),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(2.5, "mm"), legend.position = "bottom") +
  ylab("Alpha Diversity Metric") + ggtitle("Apple 2020")
line_app20
```
Plot for apple 2020, treatment on x-axis
```{r}
alpha_app20b <- alpha3 %>%
  filter(Host == "Apple", Year == "2020", Location == "CT", 
         Timepoint != "MOCK", Timepoint != "Neg", Timepoint != "resub", Timepoint != "TOSS") %>%
  mutate(Hours_post_spray = factor(Hours_post_spray, levels=c(16, 40, 64, 88),
                                   labels = c("16", "40", "64", "88"), ordered = TRUE)) %>%
  ggplot(aes(x = Treatment, y = Alpha, color = Hours_post_spray)) +
  facet_grid(Index~., scales = "free") + 
  geom_errorbar(data=line_ap20, aes(x=Treatment, y=group_mean, 
                ymin=group_mean-group_sd, ymax=group_mean+group_sd), 
                position = "dodge") +
  geom_crossbar(data=line_ap20, aes(x=Treatment, y=group_mean,  
                ymin=group_mean, ymax=group_mean),
                position = "dodge", fatten = 2) + 
  geom_jitter(position = position_jitterdodge(jitter.width=0.1), size = 2) +
  scale_color_manual(name = "Timepoint",
                     breaks = c("16", "40", "64", "88"),
                     labels = c("25% 16h","25% 40h", "100% 16h", "100% 16h"),
                     values = c("#C5E2B5", "#afa9d3", "#71C054", "#5B51A3")) +
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 8),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(3, "mm"), legend.position = "bottom") +
  ylab("Alpha Diversity Metric") + xlab(" ") #+ggtitle("Apple 2020")
alpha_app20b
```


##### Apple 2021
Plots for apple, time point on x-axis 
```{r}
# get values for error bars
line_ap21 <- alpha3 %>% 
  filter(Host == "Apple", Year == "2021", Location == "NWHRC") %>% #Hours_post_spray >=16  
  mutate(Hours_post_spray = factor(Hours_post_spray, levels=c( -8.5, 16, 40, 112, 136),
                                   labels = c( "before", "25% 16h", "25% 40h", "100% 16h", "100% 40h"),
                                              ordered = TRUE)) %>%
  group_by(Year, Host, Treatment, Timepoint, Hours_post_spray, Index) %>%
  summarise(group_mean = mean(Alpha, na.rm = TRUE), 
            group_sd = sd(Alpha, na.rm = TRUE)) %>% ungroup()

# Plots for apple, time point on x-axis 
alpha_app21 <- alpha3 %>%
  filter(Host == "Apple", Year == "2021", Location == "NWHRC", Timepoint != "0h",
         Timepoint != "MOCK", Timepoint != "Neg", Timepoint != "resub", Timepoint != "TOSS") %>%
  mutate(Hours_post_spray = factor(Hours_post_spray, levels=c(-8.5, 16, 40, 112, 136),
                                   labels = c("before", "25% 16h", "25% 40h", "100% 16h", "100% 40h"), 
                                              ordered = TRUE)) %>%
  ggplot(aes(x = Hours_post_spray, y = Alpha, color = Treatment)) +
  facet_grid(Index~., scales = "free") + 
  geom_errorbar(data=line_ap21, aes(x=Hours_post_spray, y=group_mean, 
                ymin=group_mean-group_sd, ymax=group_mean+group_sd), 
                position = "dodge") +
  geom_crossbar(data=line_ap21, aes(x=Hours_post_spray, y=group_mean,  
                ymin=group_mean, ymax=group_mean),
                position = "dodge", fatten = 2) + 
  geom_jitter(position = position_jitterdodge(jitter.width=0.1), size = 2) +
  scale_color_manual(name = "Treatment",
                     values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 10),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(3, "mm"), legend.position = "bottom") +
   ylab("Alpha Diversity Metric") + xlab("Timepoint") + ggtitle("Apple 2021")
alpha_app21

# Plots for apple, treatment on x-axis 
alpha_app21b <- alpha3 %>%
  filter(Host=="Apple" & Year=="2021" & Location=="NWHRC" &  Hours_post_spray >=16 &
         Timepoint != "MOCK", Timepoint != "Neg", Timepoint != "resub", Timepoint != "TOSS") %>%
  mutate(Hours_post_spray = factor(Hours_post_spray, ordered = TRUE,
                                   # levels=c(-8.5, 16, 40, 112, 136),
                                   # labels = c("before", "25% 16h", "25% 40h", "100% 16h", "100% 40h")
                                   levels=c(16, 40, 112, 136),
                                   labels = c("25% 16h", "25% 40h", "100% 16h", "100% 40h")
                                   )) %>%
  ggplot(aes(x = Treatment, y = Alpha, color = Hours_post_spray)) +
  facet_grid(Index~., scales = "free") + 
  geom_errorbar(data=line_ap21, aes(x=Treatment, y=group_mean, 
                ymin=group_mean-group_sd, ymax=group_mean+group_sd), position = "dodge") +
  geom_crossbar(data=line_ap21, aes(x=Treatment, y=group_mean,  
                ymin=group_mean, ymax=group_mean), position = "dodge", fatten = 2) + 
  geom_jitter(position = position_jitterdodge(jitter.width = 0.1), size = 2) +
  scale_color_manual(name = "Timepoint",
                     breaks = c( #"before",
                                "25% 16h", "25% 40h", "100% 16h", "100% 40h"),
                     labels = c( #"before",
                                "25% 16h", "25% 40h", "100% 16h", "100% 40h"),
                     values = c( #"#ED187A",
                                "#C5E2B5", "#afa9d3", "#71C054", "#5B51A3")) +
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 8),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(3, "mm"), legend.position = "bottom") +
   ylab("Alpha Diversity Metric") + xlab("Treatment") #+ ggtitle("Apple 2021")
alpha_app21b
```

# Evenness
### Format data
```{r}
H <- alpha_diversity$Shannon
S1 <- alpha_diversity$Observed
S <- log(S1)
evenness <- H/S
evenness
alpha_diversity$Evenness = evenness
alpha_diversity

# do calculation
even1 <- alpha2 %>%
  mutate(Evenness = (Shannon / log(Observed) )) %>%
  select(-Chao1, -se.chao1, -InvSimpson)
# make long format for plotting
# Make long format for plotting
even2 <- pivot_longer(data = even1, 
                       cols = c(Evenness, Observed, Shannon),
                       names_to = "Index", values_to = "Value", 
                       values_drop_na = FALSE) %>%
  mutate(Index = factor(Index, levels = c("Evenness", "Observed", "Shannon"), 
                        ordered = TRUE ))
```

### Plots
Plot evenness - Apple
```{r}
even_app <- even2 %>%
  filter(Host == "Apple", Index == "Evenness", 
         Treatment !=  "MOCK", Treatment != "Neg", Treatment != "resub", Treatment != "TOSS",
          Timepoint !=  "MOCK", Timepoint != "Neg", Timepoint != "resub", Timepoint != "TOSS",) %>%
  ggplot(aes(x = Hours_post_spray, y = Value, group = Timepoint,
             color = Year, shape = Year)) +
  facet_grid(~Treatment, scales = "free") + 
  geom_boxplot(outlier.size = -1) + 
  geom_jitter(position = position_dodge(width = 0.25), size = 2) +
  ylab("Evenness") +
  # standardize text size 
  theme(axis.text = element_text(size = 8), axis.title = element_text(size = 10), 
        strip.text = element_text(size = 10),
        legend.text = element_text(size = 6), legend.title = element_text(size=8),
        legend.key.size = unit(2.5, "mm"), legend.position = "bottom") +
  ggtitle("Apple")
even_app
```

# Bray Curtis
add meta table to sample sum data table
```{r}
sum_meta <- left_join(meta2, shared_sum,  by = "Sample_ID") %>%
  rename(SampleSum = RowSum) %>%
  select(Sample_ID, SampleSum, everything(), starts_with("OTU"))

cherry <- filter(sum_meta, Host == "Cherry", Year == "2021")

apple <- filter(sum_meta, Host == "Apple", Year == "2020")
```

### Apple 2020
```{r}
psq_ap20 <- psq %>%
  subset_samples(Host == "Apple" & Year == "2020" & Location =="CT" &
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  #subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>% #649 taxa
  prune_samples(sample_sums(.) > 3000, .) %>% #there are 2 samples below this 
  scale_reads(round = "round") #187 taxa, 24 samples

smin <- min(sample_sums(psq_ap20)) #3278
smean <- mean(sample_sums(psq_ap20)) #3287
smax <- max(sample_sums(psq_ap20)) #3295

bc_nmds3 <- ordinate(
  physeq = psq_ap20, 
  method = "NMDS", k=2, maxit=200, try=100,
  distance = "bray")
bc_nmds3 #stress= 0.181949
```

Plot in panels
```{r}
bcp3 <- phyloseq:: plot_ordination(physeq = psq_ap20,
                                   ordination = bc_nmds3,
                                   color = "Treatment") + 
  geom_point(size = 3) +
  facet_grid(Growth_stage~Timepoint, scales = "fixed") +
  scale_color_manual(name = "Treatment",values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), #reduce text size
        axis.title = element_text(size = 10), 
        plot.title = element_text(size=12),
        legend.text = element_text(size = 6),  
        legend.title = element_text(size=8), legend.justification =  "top",
        legend.key.size = unit(2.5, "mm"))  #reduce legend size
  #ggtitle("Apple 2020")
bcp3
```

Plot all together
```{r}
bcp3_b <- phyloseq:: plot_ordination(physeq = psq_ap20,
                                   ordination = bc_nmds3,
                                   color = "Treatment",
                                   shape = "Stage2") + 
  geom_point(size = 3) +
  scale_color_manual(name = "Treatment",values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), #reduce text size
        axis.title = element_text(size = 10), 
        plot.title = element_text(size=12),
        legend.text = element_text(size = 6),  
        legend.title = element_text(size=8), legend.justification =  "top",
        legend.key.size = unit(2.5, "mm")) + #reduce legend size
  ggtitle("Apple 2020")
bcp3_b
```

Statistical tests 
```{r}
psq_ap20 <- psq %>%
  subset_samples(Host == "Apple" & Year == "2020" & 
                   #Timepoint== "40h"&Growth_stage=="100% Bloom"&
                 Treatment !=  "MOCK" & Treatment != "Neg" & Treatment != "resub" & Treatment != "TOSS") %>% 
  #subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>% 
  prune_samples(sample_sums(.) > 3000, .) 

# Calculate bray curtis distance matrix
bc_bray3 <- phyloseq::distance(subset_samples(psq_ap20), method = "bray")
# make a data frame from the sample_data
bc_sampledf3 <- data.frame(sample_data(psq_ap20))
# Adonis test 
adon3 <- adonis(bc_bray3~Treatment, data = bc_sampledf3)
adon3 #p=0.053 all; 0.6 16h-25%; 0.2 40h-25%, 0.926 16h-100%; 0.6667 40h-100%

# Homogeneity of dispersion test 
beta3 <- betadisper(bc_bray3, bc_sampledf3$Treatment)
permutest(beta3) #p=0.367 all; 0.8083 16h-25%; 0.3083 40h-25%, 0.297 16h-100%; 0.8014 40h-100%
# Anosim test 
Trt <- get_variable(psq_ap20, "Treatment")
trt_ano <- anosim(bc_bray3, Trt)
trt_ano$signif #p=0.035 all; 0.6 16h-25%; 0.2 40h-25%, 0.869 16h-100%; 0.467 40h-100%
trt_ano$statistic #R= 0.136127 all; -0.25 16h-25%; 0.75 40h-25%, -0.2291667 16h-100%; 0.03571429 40h-100%
```

### Apple 2021
```{r}
psq_ap21 <- psq %>%
  subset_samples(Host == "Apple" & Year == "2021" & Location == "NWHRC" & 
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  #subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>% #649 taxa
  prune_samples(sample_sums(.) > 3000, .) %>% 
  scale_reads(round = "round") #211 taxa, 42 samples

smin <- min(sample_sums(psq_ap21)) #3147
smean <- mean(sample_sums(psq_ap21)) #3154
smax <- max(sample_sums(psq_ap21)) #3147

bc_nmds4 <- ordinate(
  physeq = psq_ap21, 
  method = "NMDS", k=2, maxit=200, try=100,
  distance = "bray")
bc_nmds4 #stress= 0.1706075
```

Plot as facets
```{r}
bcp4 <- phyloseq:: plot_ordination(physeq = psq_ap21,
                                   ordination = bc_nmds4,
                                   color = "Treatment") + 
  geom_point(size = 3) +
  facet_grid(Growth_stage~Timepoint, scales = "fixed") +
  scale_color_manual(name = "Treatment",values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), #reduce text size
        axis.title = element_text(size = 10), 
        plot.title = element_text(size=12),
        legend.text = element_text(size = 6),  
        legend.title = element_text(size=8), legend.justification =  "top",
        legend.key.size = unit(2.5, "mm")) #reduce legend size
  #ggtitle("Apple 2021")
bcp4
```

Plot all together
```{r}
bcp4_b <- phyloseq:: plot_ordination(physeq = psq_ap21,
                                   ordination = bc_nmds4,
                                   color = "Treatment",
                                   shape = "Stage2") + 
  geom_point(size = 3) +
  scale_color_manual(name = "Treatment",values = c("#FAA51A", "#426FB6")) +
  theme(axis.text = element_text(size = 8), #reduce text size
        axis.title = element_text(size = 10), 
        plot.title = element_text(size=12),
        legend.text = element_text(size = 6),  
        legend.title = element_text(size=8), legend.justification =  "top",
        legend.key.size = unit(2.5, "mm")) + #reduce legend size
  ggtitle("Apple 2021")
bcp4_b
```

Statistical tests 
```{r}
psq_ap21 <- psq %>%
  subset_samples(Host == "Apple" & Year == "2021" & Location == "NWHRC" & 
                   Timepoint== "40h"&Growth_stage=="100% Bloom"&
                 Treatment !=  "MOCK" & Treatment != "Neg" & Treatment != "resub" & Treatment != "TOSS") %>% 
  #subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") %>% 
  prune_samples(sample_sums(.) > 3000, .) 

# Calculate bray curtis distance matrix
bc_bray4 <- phyloseq::distance(subset_samples(psq_ap21), method = "bray")
# make a data frame from the sample_data
bc_sampledf4 <- data.frame(sample_data(psq_ap21))
# Adonis test 
adon4 <- adonis(bc_bray4~Treatment, data = bc_sampledf4)
adon4 #p=0.407 all; 0.007 0h-25%; 0.1 16h-25%; 0.188 40h-25%, 0.191 16h-100%; 0.054 40h-100%

# Homogeneity of dispersion test 
beta4 <- betadisper(bc_bray4, bc_sampledf4$Treatment)
permutest(beta4) #p=0.289 all; 0.107 0h-25%; 0.9083 16h-25%; 0.097 40h-25%, 0.044 16h-100%; 0.558 40h-100%
# Anosim test 
Trt <- get_variable(psq_ap21, "Treatment")
trt_ano <- anosim(bc_bray4, Trt)
trt_ano$signif #p=0.519 all; 0.004 0h-25%; 0.1 16h-25%; 0.296 40h-25%, 0.431 16h-100%; 0.049 40h-100%
trt_ano$statistic #R= -0.00643559 all; 0.45625 0h-25%; 1.0 16h-25%; 0.03125 40h-25%, -0.0125 16h-100%; 0.404 40h-100%
```

# DeSeq
Format data
```{r}
library(DESeq2)
alpha <- 0.05
# transform count data to +1 to every cell for DeSeq (no zeros)
psq_n <- column_to_rownames(shared2, var = "Sample_ID") %>%
  mutate_all(funs(.+1)) %>%
  otu_table(., taxa_are_rows = FALSE) %>%
  phyloseq(., tax_table(as.matrix(rdp)),
           sample_data(column_to_rownames(meta2, var = "Sample_ID")))
# Apple 2020
psq_ap20 <- psq_n %>%
  subset_samples(Host == "Apple" & Year == "2020" & Location == "CT",
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") 
# Apple 2021
psq_ap21 <- psq_n %>%
  subset_samples(Host == "Apple" & Year == "2021" & Location == "NWHRC" &
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") 
# Cherry 2020
psq_ch20 <- psq_n %>%
  subset_samples(Host == "Cherry" & Year == "2020" & Location == "EL" &
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") 
# Cherry 2021 
psq_ch21 <- psq_n %>%
  subset_samples(Host == "Cherry" & Year == "2021" & Location == "EL" &
                 Treatment !=  "MOCK" & Treatment != "Neg" & 
                   Treatment != "resub" & Treatment != "TOSS") %>% 
  subset_taxa(Family != "Chloroplast" & Phylum != "Plantae") 
```

### Apple 2020
```{r}
# convert phyloesq obj to deseq and do the calculations
ddsap20 <- DESeq(phyloseq_to_deseq2(psq_ap20, ~Treatment))
# extract results table, set p-value, variables to compare 
ddsap20$Treatment <- relevel(ddsap20$Treatment, ref = "Water")
dds3 <- results(ddsap20, alpha=0.05, 
                contrast=c("Treatment", "Water", "Agriphage")) #factor, numerator, denominator
summary(dds3) 
# subset this table to specified significance level
sigtab_dds3 <- dds3[which(dds3$padj < 0.05), ]
# significantly differentially abundant
summary(sigtab_dds3) 
# combine with the OTU taxonomy table
sigtab_dds3 <- cbind(as(sigtab_dds3, "data.frame"), 
                     as(tax_table(psq_ap20)[row.names(sigtab_dds3), ], "matrix"))
# sort by baseMean column
sigtab_dds3[order(sigtab_dds3$baseMean, decreasing=T), ]
```

Plot taxa with different abundance after agriphage treatment compared to water-treated. 
```{r}
# Phylum order
x <- tapply(sigtab_dds3$log2FoldChange, sigtab_dds3$Phylum, function(x) max(x))
x <- sort(x, TRUE)
sigtab_dds3$Phylum <- factor(as.character(sigtab_dds3$Phylum), levels=names(x))

# Genus order
x <- tapply(sigtab_dds3$log2FoldChange, sigtab_dds3$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_dds3$Genus <- factor(as.character(sigtab_dds3$Genus), levels=names(x))

plot_ds3 <- sigtab_dds3 %>%
  #dplyr::mutate(Genus = str_replace_na(Genus, replacement = "NA")) %>%
  #filter(Genus != "NA") %>%
  ggplot(aes(x=Genus, y=log2FoldChange, color=Phylum)) + 
  geom_hline(yintercept = 0.0, color = "gray", size = 0.5) +
  geom_point(size=6) + ggtitle("Apple 2020") +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) 
plot_ds3
```

### Apple 2021
```{r}
# convert phyloesq obj to deseq and do the calculations
ddsap21 <- DESeq(phyloseq_to_deseq2(psq_ap21, ~Treatment))
# extract results table, set p-value, variables to compare 
ddsap21$Treatment <- relevel(ddsap21$Treatment, ref = "Water")
dds4 <- results(ddsap21, alpha=0.05, 
                contrast=c("Treatment", "Water", "Agriphage") )
summary(dds4) 
# subset this table to specified significance level
sigtab_dds4 <- dds4[which(dds4$padj < 0.05), ]
# significantly differentially abundant
summary(sigtab_dds4) 
# combine with the OTU taxonomy table
sigtab_dds4 <- cbind(as(sigtab_dds4, "data.frame"), 
                     as(tax_table(psq_ap21)[row.names(sigtab_dds4), ], "matrix"))
# sort by baseMean column
sigtab_dds4[order(sigtab_dds4$baseMean, decreasing=T), ]
```

Plot taxa with different abundance after agriphage treatment compared to water-treated. 
```{r}
# Phylum order
x <- tapply(sigtab_dds4$log2FoldChange, sigtab_dds4$Phylum, function(x) max(x))
x <- sort(x, TRUE)
sigtab_dds4$Phylum <- factor(as.character(sigtab_dds4$Phylum), levels=names(x))

# Genus order
x <- tapply(sigtab_dds4$log2FoldChange, sigtab_dds4$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_dds4$Genus <- factor(as.character(sigtab_dds4$Genus), levels=names(x))

plot_ds4 <- sigtab_dds4 %>%
  mutate(Genus = str_replace_na(Genus, replacement = "NA")) %>%
  filter(Genus != "NA") %>%
  ggplot(aes(x=Genus, y=log2FoldChange, color=Phylum)) + 
  geom_hline(yintercept = 0.0, color = "gray", size = 0.5) +
  geom_point(size=6) + ggtitle("Apple 2021") +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) 
plot_ds4
```

# Save plots
```{r}
ggsave2(plot = alpha_app20, filename = "figures/alpha_apple_2020.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)
ggsave2(plot = alpha_app21, filename = "figures/alpha_apple_2021.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)

ggsave2(plot = area1, filename = "figures/area_apple_2020.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)
ggsave2(plot = area2, filename = "figures/area_apple_2021.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)

ggsave2(plot = plot_ds3, filename = "figures/deseq_apple_2020.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)
ggsave2(plot = plot_ds4, filename = "figures/deseq_apple_2021.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)

ggsave2(plot = bcp3+theme(legend.position = "none"), filename = "figures/nmds_apple_2020.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)
ggsave2(plot = bcp4+theme(legend.position = "none"), filename = "figures/nmds_apple_2021.tif", 
        device = "tiff", units = "mm", width = 178, height = 178, dpi = 300)
```

# Multipanel figure
```{r}
# shows diversity increases after phage application: alpha_cher20b 
leg1 <- get_legend(alpha_app20b)
top <- plot_grid(alpha_app20b+theme(legend.position="none"), 
                 alpha_app21b+theme(legend.position="none"),
                 nrow=1, ncol=2, labels = c("A", "B"))
top2 <- plot_grid(top, leg1, 
                  nrow = 2, ncol=1,
                  rel_heights = c(1, 0.1), axis="l", align="hv")
leg2 <- get_legend(area1)
bottom <- plot_grid(area1+theme(legend.position="none"), 
                    area2+theme(legend.position="none"), 
                    leg2, labels = c("C", "D"),
                    nrow = 1, ncol = 3, rel_widths = c(1, 1, 0.5))
mp_test <- plot_grid(top2, bottom, nrow = 2, ncol = 1) 

ggsave2(plot = mp_test, filename = "figures/phage_apple_multipanel.tif", 
        device = "tiff", units = "mm", width = 356, height = 178, dpi = 300)
```


# Export data
Alpha diversity indices
```{r}
# raw values
write_delim(alpha2, file = "curated_data/alpha_stats.csv",
            delim = ",", col_names = TRUE)
# means of replicates 
write_delim( 
  rbind(line_ap20 %>% # apple 2020 sum stats
          pivot_wider(names_from = "Index", 
                      values_from = c("group_mean", "group_sd")), 
        line_ap21 %>% # apple 2021 summary stats
          pivot_wider(names_from = "Index", values_from = c("group_mean", "group_sd"))
  ),
  file = "curated_data/alpha_apples.csv",
  delim = ",", col_names = TRUE)

# plant pathogen genus table
write_delim(gen_tab, file = "pathogen_genera_table.csv",
             delim = ",", col_names = TRUE)

gen_tab %>%
  group_by(Location, Treatment, Growth_stage, Timepoint) %>%
  summarise(Total = sum(Avg_abund, na.rm = TRUE)) %>%
  write_delim(., file = "pathogen_genera_table2.csv",
              delim = ",", col_names = TRUE)

write_delim(gen_rep_tab, file = "pathogen_genera_table3.csv",
             delim = ",", col_names = TRUE)
```

-----
end
